pkg/apis/core/validation/validation.go

```diff
@@ -7466,7 +7466,7 @@ func validateEndpointAddress(address *core.EndpointAddress, fldPath *field.Path)
 	// During endpoint update, verify that NodeName is a DNS subdomain and transition rules allow the update
 	if address.NodeName != nil {
 		for _, msg := range ValidateNodeName(*address.NodeName, false) {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("nodeName"), *address.NodeName, msg))
+			allErrs.AppendWithOrigin("NodeName", field.Invalid(fldPath.Child("nodeName"), *address.NodeName, msg))
```

Comment:
This is an interesting case.  I think the origin should be the same as the function underlying NodeName (which was given a name alias to hide the details).  Think of what we would put here in the case of a declarative tag, it would be +subfield(name)=+format=dns-label' or something?

I forget which one we use for Node, and am on mobile ATM.